src/test/java/org/tugraz/sysds/test/functions/builtin/BuiltinSTEPLmTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2018 Graz University of Technology
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tugraz.sysds.test.functions.builtin;
+
+import org.junit.Test;
+import org.tugraz.sysds.common.Types.ExecMode;
+import org.tugraz.sysds.lops.LopProperties.ExecType;
+import org.tugraz.sysds.runtime.matrix.data.MatrixValue.CellIndex;
+import org.tugraz.sysds.test.AutomatedTestBase;
+import org.tugraz.sysds.test.TestConfiguration;
+import org.tugraz.sysds.test.TestUtils;
+
+import java.util.HashMap;
+
+
+public class BuiltinSTEPLmTest extends AutomatedTestBase{
+
+    private final static String TEST_NAME = "steplm";
+    private final static String TEST_DIR = "functions/builtin/";
+    private static final String TEST_CLASS_DIR = TEST_DIR + BuiltinSTEPLmTest.class.getSimpleName() + "/";
+
+    private final static double eps = 1e-10;
+    private final static int rows = 10;
+    private final static int cols = 3;
+    private final static double spSparse = 0.3;
+    private final static double spDense = 0.7;
+
+    public enum LinregType {
+        CG, DS, AUTO
+    }
+
+    @Override
+    public void setUp() {
+        addTestConfiguration(TEST_NAME,new TestConfiguration(TEST_CLASS_DIR, TEST_NAME,new String[]{"B"}));
+    }
+
+    @Test
+    public void testLmMatrixDenseCPlm() {
+        runLmTest(false, ExecType.CP, BuiltinLmTest.LinregType.AUTO);
+    }
+
+    @Test
+    public void testLmMatrixDenseSPlm() {
+        runLmTest(false, ExecType.SPARK, BuiltinLmTest.LinregType.AUTO);
+    }
+
+    @Test
+    public void testLmMatrixSparseSPlm() {
+        runLmTest(true, ExecType.SPARK, BuiltinLmTest.LinregType.AUTO);
+    }
+
+    @Test
+    public void testLmMatrixDenseCPlmDS() {
+        runLmTest(false, ExecType.CP, BuiltinLmTest.LinregType.DS);
+    }
+
+    @Test
+    public void testLmMatrixSparseCPlmDS() {
+        runLmTest(true, ExecType.CP, BuiltinLmTest.LinregType.DS);
+    }
+
+    @Test
+    public void testLmMatrixDenseSPlmDS() {
+        runLmTest(false, ExecType.SPARK, BuiltinLmTest.LinregType.DS);
+    }
```

Comment:
Remove all unnecessary testcases - as in steplm you don't differentiate between DS, CG, or AUTO